data/campaigns/Heir_To_The_Throne/scenarios/05b_Isle_of_the_Damned.cfg

```diff
@@ -34,6 +34,95 @@
         [/objectives]
     [/event]
 
+    # This appoints the Advisor role to our most advanced unit coming out of The Bay of Pearls. This would be
+    # the same as the initial Advisor we would appoint if we took the path through Muff Malal's Peninsula, and
+    # will be the initial Advisor recalled at the start of The Siege of Elensefar.
+    #
+    # We create a shallow copy of this Advisor, needing only the name, type and profile image. The shallow copy
```

Comment:
There is a ship unit type, which is why I suggested making the ship the speaker - I'd assumed a unit was already being used for the ship (didn't look too closely, I guess). There's also a key in [message] which overrides the displayed speaker name (I think it's `caption`, or maybe `caption_text`; I'm pretty sure a string for `speaker` doesn't do what you seem to want it to do). Thus you could in theory use a generic ship, but override the name and image to look like the chosen unit.

If you're not using a ship unit though (maybe the ship is just an `[item]`?), I can understand why you would've done it the way you did. If the ship is not a unit, then the hidden speaker can be on the same space as the ship.

I don't know which way is better, really. Whatever works, I guess.